src/algorithms/optimization/fixed_point_differentiation.jl

```diff
@@ -279,8 +280,18 @@ end
 
 function _fix_svd_algorithm(alg::SVDAdjoint, signs, info)
     # embed gauge signs in larger space to fix gauge of full U and V on truncated subspace
-    signs_full = map(zip(signs, info.S_full)) do (σ, S_full)
-        extended_σ = zeros(scalartype(σ), space(S_full))
+    signs_full = map(Iterators.product(1:4, 1:size(signs, 2), 1:size(signs, 3))) do (dir, r, c)
+        σ = signs[dir, r, c]
+        extended_space = if dir == NORTH # take unit cell interdependency of signs into account
```

Comment:
This might read slightly easier if you do something like 
```julia
r_ext, c_ext = if ...

else
...
end
extended_space = domain(info.U_full[dir, r_ext, c_ext]) \leftarrow codomain(info.V_full[dir, r_ext, c_ext])
```